src/Centreon/Application/Controller/PlatformTopologyController.php

```diff
@@ -0,0 +1,193 @@
+            . 'config/json_validator/latest/Centreon/PlatformTopology/AddServer.json'
+        );
+
+        // sanitize data
+        $platformToAdd['ip_address'] = filter_var($platformToAdd['ip_address'], FILTER_VALIDATE_IP);
+        $platformToAdd['server_name'] = filter_var($platformToAdd['server_name'], FILTER_SANITIZE_STRING);
+        $platformToAdd['server_parent'] = !empty($platformToAdd['server_parent'])
+            ? filter_var($platformToAdd['server_parent'], FILTER_VALIDATE_IP)
+            // if not server_parent is sent, then adding it to the central
+            : $_SERVER['SERVER_ADDR'];
+
+        if (empty($platformToAdd['server_name'])) {
+            return $this->view([
+                'code' => Response::HTTP_BAD_REQUEST,
+                'message' => _('The name of the platform is not consistent')
+            ]);
+        }
+
+        if (false === $platformToAdd['ip_address']) {
+            return $this->view([
+                'code' => Response::HTTP_BAD_REQUEST,
+                'message' => sprintf(
+                    _("The address of the platform '%s' is not consistent"),
+                    $platformToAdd['server_name']
+                )
+            ]);
+        }
+
+        // check server type consistency
+        if (0 === $platformToAdd['server_type']) {
+            return $this->view([
+                'code' => Response::HTTP_BAD_REQUEST,
+                'message' => sprintf(
+                    _("You cannot link the Central '%s'@'%s' to another Central"),
+                    $platformToAdd['server_name'],
+                    $platformToAdd['ip_address']
+                )
+            ]);
+        } elseif (!isset(static::ALLOWED_TYPES[$platformToAdd['server_type']])) {
+            return $this->view([
+                'code' => Response::HTTP_BAD_REQUEST,
+                'message' => sprintf(
+                    _("The type of platform '%s'@'%s' is not consistent"),
+                    $platformToAdd['server_name'],
+                    $platformToAdd['ip_address']
+                )
+            ]);
+        }
+
+        if (false === $platformToAdd['server_parent']) {
+            return $this->view([
+                'code' => Response::HTTP_BAD_REQUEST,
+                'message' => sprintf(
+                    _("The address of the parent platform '%s'@'%s' is not consistent"),
+                    $platformToAdd['server_name'],
+                    $platformToAdd['ip_address']
+                )
+            ]);
+        }
+
+        $setPlatformTopology = (new PlatformTopology())
+            ->setServerAddress($platformToAdd['ip_address'])
+            ->setServerName($platformToAdd['server_name'])
+            ->setserverType($platformToAdd['server_type'])
+            ->setServerParentAddress($platformToAdd['server_parent']);
+
+        $this->platformTopologyService->addPlatformToTopology($setPlatformTopology);
+
+        return $this->view([
```

Comment:
no need to return a message when the resource has been created correctly

Reply:
@jeremyjaouen : As discussed. You'll need to generate the message from the request status code